lib/bundler/templates/newgem/bin/setup.tt

```diff
@@ -1,5 +1,6 @@
 #!/usr/bin/env bash
 set -euo pipefail
+cd -
```

Comment:
Because shells like MacOS Finder produce the same problem to setup as well. They execute it with ~ as active directory and bundle install command just can't find proper files.

With this line I say to script "please, set the directory you're in as the current one". I've found "cd -" one-liner a long time ago somewhere in bash manuals.